lib/ansible/modules/cloud/vmware/vmware_object_rename.py

```diff
@@ -0,0 +1,172 @@
+     required: True
+   object_type:
+     description:
+     - Type of object to rename.
+     required: False
+     default: 'VirtualMachine'
+     choices: ['VirtualMachine', 'Datacenter', 'ClusterComputeResource',
+               'DistributedVirtualSwitch', 'HostSystem', 'Datastore',
+               'Network', 'DistributedVirtualPortgroup']
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: Rename Datacenter
+  vmware_object_rename:
+    hostname: 192.168.1.209
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: False
+    name: datacenter1
+    new_name: datacenter2
+    object_type: 'Datacenter'
+  delegate_to: localhost
+
+- name: Rename Cluster
+  vmware_object_rename:
+    hostname: 192.168.1.209
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: False
+    name: cluster1
+    new_name: nw_cluster2
+    object_type: 'ClusterComputeResource'
+  delegate_to: localhost
+'''
+
+RETURN = """ # """
+
+try:
+    import pyVmomi
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task
+
+
+class VmObjectMgr(PyVmomi):
+    def __init__(self, module):
+        super(VmObjectMgr, self).__init__(module)
+
+        object_types = {
+            'Datacenter': vim.Datacenter,
+            'VirtualMachine': vim.VirtualMachine,
+            'ClusterComputeResource': vim.ClusterComputeResource,
+            'DistributedVirtualSwitch': vim.DistributedVirtualSwitch,
+            'HostSystem': vim.HostSystem,
+            'Datastore': vim.Datastore,
+            'Network': vim.Network,
+            'DistributedVirtualPortgroup': vim.dvs.DistributedVirtualPortgroup,
+        }
+
+        old_name = module.params['name']
+        new_name = module.params['new_name']
+        object_type = object_types[module.params['object_type']]
+
+        managed_objects = self.get_managed_objects_properties(vim_type=object_type, properties=['name'])
+
+        objects_list = []
+        already_exists = False
+
+        for m_obj in managed_objects:
+            if m_obj.obj.name == old_name:
+                objects_list.append(m_obj.obj)
+            if m_obj.obj.name == new_name:
+                already_exists = True
+
+        if len(objects_list) > 1:
+            self.module.fail_json(msg="Multiple objects found with the same name %s."
+                                      " Please make sure you have unique name to use this feature." % old_name)
+
+        if already_exists:
+            self.module.fail_json(msg="Unable to rename object %s as object with"
+                                      " same name %s already exists." % (old_name, new_name))
+
+        if module.check_mode:
+            self.module.exit_json(changed=True)
+
+        if objects_list:
```

Comment:
no need for this "if" if you apply the previous change.

Reply:
and you can remove the changed=False section